### PR TITLE
feat(Menu): add support for item icon on the end of the item

### PR DESCRIPTION
**Pull request by ogonkov**

### CODE REVIEW

**Review by krozhkov — APPROVED** [review]

File: src/components/Menu/__stories__/Menu.stories.tsx [hunk 1]
```diff
@@ -29,7 +29,7 @@ export const ItemActive: StoryFn<MenuProps> = (args) => (
 
 export const ItemIcon: StoryFn<MenuProps> = (args) => (
     <Menu {...args}>
-        <Menu.Item icon={<Icon data={Gear} size={16} />}>Settings</Menu.Item>
+        <Menu.Item iconStart={<Icon data={Gear} size={16} />}>Settings</Menu.Item>
```
Comment (NikitaCG) [inline src/components/Menu/__stories__/Menu.stories.tsx:32]: please add case with iconEnd

File: src/components/Menu/MenuItem.tsx [hunk 2]
```diff
@@ -7,7 +7,10 @@ import {eventBroker} from '../utils/event-broker';
 const b = block('menu');
 
 export interface MenuItemProps extends DOMProps, QAProps {
+    /** @deprecated use `iconStart` instead */
     icon?: React.ReactNode;
+    iconStart?: React.ReactNode;
+    iconEnd?: React.ReactNode;
```
Comment (NikitaCG) [inline src/components/Menu/MenuItem.tsx:13]: Please refactor README.md. We are need new props description and cases for landing and github parts. https://github.com/gravity-ui/uikit/blob/main/src/components/Menu/README.md#icon

**Review by NikitaCG — CHANGES_REQUESTED** [review]

File: src/components/Menu/README.md [hunk 3]
```diff
@@ -139,13 +139,13 @@ Used if you need to display an icon for a menu item
...
 <Menu>
-    <Menu.Item icon={<Icon size={16} data={GearIcon} />}>Item with icon</Menu.Item>
+    <Menu.Item iconStart={<Icon size={16} data={GearIcon} />}>Item with icon</Menu.Item>
     <Menu.Item>Item without icon</Menu.Item>
 </Menu>
 `}
 >
     <UIKit.Menu>
-        <UIKit.Menu.Item icon={
+        <UIKit.Menu.Item iconStart={
```
Comment (NikitaCG) [inline src/components/Menu/README.md:148]: please add showcase with iconEnd

**Review by NikitaCG — APPROVED** [review]

File: src/components/Menu/README.md (hunk 3 above)
Comment (ogonkov) [reply src/components/Menu/README.md:148]: Done

**Review by korvin89 — APPROVED** [review]

File: src/components/DropdownMenu/DropdownMenuItem.tsx [hunk 4]
```diff
@@ -92,20 +92,22 @@ export const DropdownMenuItem = <T,>({
         };
     }, [props.extraProps, closeSubmenu, hasSubmenu, openSubmenu]);
 
+    let iconEnd = props.iconEnd ? props.iconEnd : null;
```
[the remark this answers is not in the document]
Comment (ogonkov) [reply src/components/DropdownMenu/DropdownMenuItem.tsx:95]: Done

[the remark this answers is not in the document]
Comment (axtk) [reply src/components/DropdownMenu/DropdownMenuItem.tsx:95]: > Done

Is it?

File: src/components/Menu/MenuItem.tsx [hunk 5]
```diff
@@ -8,7 +8,10 @@ import {eventBroker} from '../utils/event-broker';
 const b = block('menu');
 
 export interface MenuItemProps extends DOMProps, QAProps {
+    /** @deprecated use `iconStart` instead */
```
Comment (axtk) [inline src/components/Menu/MenuItem.tsx:11]: With this change, the `DropdownMenu`'s README also requires an update, at least in the sections *Custom icons* and *DrowdownMenuItem*

Comment (ogonkov) [reply src/components/Menu/MenuItem.tsx:11]: Done

File: src/components/DropdownMenu/README.md [hunk 6]
```diff
@@ -399,7 +399,7 @@ The example above is oversimplified to demonstrate the idea of the customizable
 
 ## Custom icons
 
-A custom icon can be added to a `DropdownMenu` item by assigning the `icon` property. By default, `DropdownMenu` items go without icons.
+A custom icons can be added to a `DropdownMenu` item by assigning the `iconStart` or `iconEnd` property. By default, `DropdownMenu` items go without icons.
```
Comment (axtk) [inline src/components/DropdownMenu/README.md:402]: Let's replace 'A custom icons' with 'Custom icons'

File: src/components/DropdownMenu/README.md [hunk 7]
```diff
@@ -528,7 +528,8 @@ This type describes individual dropdown menu items.
 | :----------- | :------------------------------------------------------------------------------------------------------------------------------ | :--------------------------------------------: | :-----: |
 | `text`       | Menu item content.                                                                                                              |               `React.ReactNode`                |         |
 | `action`     | Menu item click handler. Recieves the parameters from the parent dropdown menu component (both `event` and `data`).             | `(event: React.MouseEvent, data: any) => void` |         |
-| `icon`       | Menu item icon.                                                                                                                 |               `React.ReactNode`                |         |
+| `iconStart`  | Menu item icon before item text.                                                                                                |               `React.ReactNode`                |         |
```
Comment (axtk) [inline src/components/DropdownMenu/README.md:531]: Let's rephrase it as 'Menu item icon before the item content.'

File: src/components/DropdownMenu/README.md [hunk 8]
```diff
@@ -528,7 +528,8 @@ This type describes individual dropdown menu items.
...
+| `iconEnd`    | Menu item icon after item text. Ignored, when item has sub-menu.                                                                |               `React.ReactNode`                |         |
```
Comment (axtk) [inline src/components/DropdownMenu/README.md:532]: &rarr; 'Menu item icon after the item content. Ignored if the item has a submenu.'

**Review by axtk — APPROVED** [review]

**Pull request merged**